sdlib/d/gc/region.d

```diff
@@ -352,13 +352,37 @@ unittest enormous {
 }
 
 struct Region {
-	void* address;
+	/**
+	 * This is a bitfield containing the following elements:
+	 *  - c: The alloc class.
+	 *  - a: The address (BlockSize-aligned)
+	 *  - g: The generation.
+	 * 
+	 * 63    56 55    48 47    40 39    32 31    24 23    16 15     8 7      0
+	 * ........ cccccccc aaaaaaaa aaaaaaaa aaaaaaaa aaa..... ........ gggggggg
+	 */
+	ulong bits;
+
+	// Verify our assumptions.
+	static assert(LgAddressSpace <= 48, "Address space too large!");
```

Comment:
You could do with 56 bits here. Not that it changes much in practice, but why pessimise?